datafusion/core/tests/sql/select.rs

```diff
@@ -350,3 +350,45 @@ async fn test_version_function() {
 
     assert_eq!(version.value(0), expected_version);
 }
+
+#[tokio::test]
+async fn test_subscript() -> Result<()> {
```

Comment:
This is more suitable as a sqllogictest test. Maybe we can move it to [struct.slt](https://github.com/apache/datafusion/blob/main/datafusion/sqllogictest/test_files/struct.slt) or [identifiers.slt](https://github.com/apache/datafusion/blob/main/datafusion/sqllogictest/test_files/identifiers.slt)